File: trees/Height.py
#!/usr/bin/env python
# coding: utf-8

# In[9]:


class Node: 
  
    # Constructor to create a new node 
    def __init__(self, data): 
        self.data = data 
        self.left = None
        self.right = None

def height(node): 
    if node is None: 
        return 0 ;  
  
    else : 
        left_height = height(node.left) 
        right_height = height(node.right) 

        # Use the larger one 
        if (left_height > right_height): 
            return left_height+1
        else: 
            return right_height+1

# Driver code 
root = Node(1) 
root.left = Node(2) 
root.right = Node(3) 
root.left.left = Node(4) 
root.left.right = Node(5) 
  
print ("Height of tree is %d" %(height(root))) 


# In[ ]:

File: trees/height/Clang/height.c
#include <stdio.h>

typedef struct s_node {
  int data;
  struct s_node* left;
  struct s_node* right;
} Node;

int height(Node* node) {
  if(node == NULL) {
    return 0;
  }
  int heightLeft = height(node->left);
  int heightRight = height(node->right);

  if(heightLeft > heightRight) {
    return heightLeft + 1;
  }
  return heightRight + 1;
}


int main(void) {
  Node left_left = {4, NULL, NULL};
  Node left_right = {5, NULL, NULL};
  Node left = {2, &left_left, &left_right};
  Node right = {3, NULL, NULL};
  Node root = {1, &left, &right};

  // Tree Structure is as follows:
  //            1 
  //          🡧  🡦
  //        2      3
  //      🡧  🡦
  //    4       5
  // Height = 3

  printf("Height of tree is %d", height(&root));
}
